manager/integration/tests/test_migration.py

```diff
@@ -225,22 +232,135 @@ def test_migration_with_rebuilding_replica(clients, volume_name):  # NOQA
+    # Step 1
+    lht_host_id = get_self_host_id()
+    volume = create_and_check_volume(client,
+                                     volume_name,
+                                     REPLICA_COUNT,
+                                     size=str(DEFAULT_VOLUME_SIZE * Gi))
+
+    volume.attach(hostId=lht_host_id)
+    volume = common.wait_for_volume_healthy(client, volume_name)
+
+    bv, b, _, _ = create_backup(client, volume_name)
+
+    # Step 2
+    storage_class['metadata']['name'] = "longhorn-from-backup"
+    storage_class['parameters']['fromBackup'] = b.url
+    storage_class['parameters']['migratable'] = "true"
+
+    create_storage_class(storage_class)
+
+    # Stpe 3
+    VOLUME_SIZE = str(DEFAULT_VOLUME_SIZE * Gi)
+
+    backup_pvc_name = generate_volume_name()
+
+    backup_pvc_spec = {
+        "apiVersion": "v1",
+        "kind": "PersistentVolumeClaim",
+        "metadata": {
+                "name": backup_pvc_name,
+        },
+        "spec": {
+            "accessModes": [
+                "ReadWriteMany"
+            ],
+            "storageClassName": storage_class['metadata']['name'],
+            "resources": {
+                "requests": {
+                    "storage": VOLUME_SIZE
+                }
+            }
+        }
+    }
+
+    volume_count = len(client.list_volume())
+
+    core_api.create_namespaced_persistent_volume_claim(
+        'default',
+        backup_pvc_spec
+    )
+
+    backup_volume_created = False
+
+    for i in range(RETRY_COUNTS):
+        if len(client.list_volume()) == volume_count + 1:
+            backup_volume_created = True
+            break
+        time.sleep(RETRY_INTERVAL)
+
+    assert backup_volume_created
+
+    for i in range(RETRY_COUNTS):
+        pvc_status = core_api.read_namespaced_persistent_volume_claim_status(
+            name=backup_pvc_name,
+            namespace='default'
+        )
+
+        if pvc_status.status.phase == 'Bound':
+            break
+        time.sleep(RETRY_INTERVAL)
+
+    found = False
+    for i in range(RETRY_COUNTS):
+        volumes = client.list_volume()
+        for volume in volumes:
+            if volume.kubernetesStatus.pvcName == backup_pvc_name:
+                backup_volume_name = volume.name
+                found = True
+                break
+        if found:
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert found
+
+    wait_for_volume_restoration_completed(client, backup_volume_name)
+    wait_for_volume_detached(client, backup_volume_name)
+
+    # Step 4
+    current_node = common.get_self_host_id()
+    host1, host2 = get_hosts_for_migration_test(clients)
+
+    volume = client.by_id_volume(backup_volume_name)
+    volume.attach(hostId=current_node)
+    volume = common.wait_for_volume_healthy(client, backup_volume_name)
+
+    data = common.write_volume_random_data(volume)
+    volume.detach(hostId="")
+    volume = common.wait_for_volume_detached(client, backup_volume_name)
+
+    # Step 5, 6
```

Comment:
Did you check `requestedBackupRestore`? Please make sure following the steps above, otherwise we will miss some validation.

Or if the validation above is not right, you can update the test description as well. Otherwise, should implement the way as described above.

Reply:
Sorry I missed this part, I update code for review, thank you.